Source/CarthageKit/Xcode.swift

```diff
@@ -733,26 +761,31 @@ private func shouldBuildProductType(productType: ProductType) -> Bool {
 	return productType == .Framework
 }
 
+/// Determines whether the specified Mach-O type should be built automatically.
+private func shouldBuildMachOType(machOType: MachOType) -> Bool {
+	return machOType == .Dylib
+}
+
 /// Determines whether the given scheme should be built automatically.
 private func shouldBuildScheme(buildArguments: BuildArguments, _ forPlatforms: Set<Platform>) -> SignalProducer<Bool, CarthageError> {
 	precondition(buildArguments.scheme != nil)
 
 	return BuildSettings.loadWithArguments(buildArguments)
-		.flatMap(.Concat) { settings -> SignalProducer<ProductType, CarthageError> in
-			let productType = SignalProducer(result: settings.productType)
+		.flatMap(.Concat) { settings -> SignalProducer<(ProductType, MachOType), CarthageError> in
+			let typePair = SignalProducer(result: settings.productType &&& settings.machOType)
 
 			if forPlatforms.isEmpty {
-				return productType
+				return typePair
 					.flatMapError { _ in .empty }
 			} else {
 				return settings.buildSDKs
 					.filter { forPlatforms.contains($0.platform) }
-					.flatMap(.Merge) { _ in productType }
+					.flatMap(.Merge) { _ in typePair }
 					.flatMapError { _ in .empty }
 			}
 		}
-		.filter(shouldBuildProductType)
-		// If we find any framework target, we should indeed build this scheme.
+		.filter { shouldBuildProductType($0) && shouldBuildMachOType($1) }
```

Comment:
I think we should either (1) combine these into one function that takes a product type _and_ Mach-O type or (2) combine the two enums into a single enum that differentiates between dynamic and static frameworks.

I'd lean towards (2) but could be good with (1) if we think it's important for build setting properties to have a 1-to-1 correspondence.